esx_service/vmdk_ops.py

```diff
@@ -619,19 +606,34 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
 
     return response
 
-def connectLocal():
+def connectLocalSi(force=False):
     '''
-	connect and do stuff on local machine
+	Initialize a connection to the local SI
 	'''
-    global si  #
-    # Connect to localhost as dcui
-    # User "dcui" is a local Admin that does not lose permissions
-    # when the host is in lockdown mode.
-    si = pyVim.connect.Connect(host='localhost', user='dcui')
-    if not si:
-        raise SystemExit("Failed to connect to localhost as 'dcui'.")
-
-    atexit.register(pyVim.connect.Disconnect, si)
+    global _service_instance
+    if not _service_instance:
+        try:
+            logging.info("Connecting to the local Service Instance")
+            _service_instance = pyVim.connect.Connect(host='localhost', user='dcui')
+        except Exception as e:
+            logging.exception("Failed to the local Service Instance as 'dcui': ")
+            sys.exit(1)
+    elif force:
+        logging.warning("Reconnecting to the local Service Instance")
+        _service_instance = pyVim.connect.Connect(host='localhost', user='dcui')
+
+    atexit.register(pyVim.connect.Disconnect, _service_instance)
+
+def get_si():
+    '''
+	Return a connection to the local SI
+	'''
+    with getLock('siLock'):
+        try:
+            _service_instance.CurrentTime()
```

Comment:
Will CurrentTime() indeed verify validity of _service_instance or will it return cached value? I am not an expert and [documentation](http://pubs.vmware.com/vi3/sdk/ReferenceGuide/vim.ServiceInstance.html#currentTime) isn't clear either. I wonder if we should be using RetrieveServiceContent to make sure _service_instance is still valid?